### PR TITLE
Refactor Actions Page and Sagas

### PR DESCRIPTION
**Pull request by rdig**
This PR refactors the `ActionsPage` component to be able to support custom action components per different types of actions.

Note that this does not add all the specific components for all the actions we currently have, just adds the framework for us to be able to do it down the line.

The first "custom" component for this will be added after I rebase the changes from here into #2373

**Changes** 

- [x] Added `DefaultAction` sub-component
- [x] Refactored `ActionsPage` to support custom components for specific actions
- [x] Refactored all actions sagas into their own files

Incidentally also fixed the following bugs that were reported in the notion doc:
- https://www.notion.so/colony/Dapp-Launch-QA-435e910d2dcc4fe799754d12e67e0689#6fcb7e486fe74bcc83a7f2fc87dfcc16 
- https://www.notion.so/colony/Dapp-Launch-QA-435e910d2dcc4fe799754d12e67e0689#280baa15c6bc48438e079aa5c313dea4

Resolves DEV-216

### CODE REVIEW

**Review by alicjakujawa — APPROVED** [review]
Awesome first step to make ActionPage better 🔥

**Review by chinins — APPROVED** [review]
Works & looks good!

**Comment by rdig** [comment]
@alicjakujawa @chinins 

I've added some new stuff to this refactor, I've split up the actions saga file into separate files for each action, as the initial one got over 1000 lines and was beginning to be hard to read.

I'd appreciate another look-over when you have the time.

**Review by chinins — APPROVED** [review]
Looks good. I made a comment about changing pots to teams in the translation en.json file.

Also, I didn't understand what the bug was and how you fixed it. What am I missing?

Screenshot 2021-02-19 at 11 00 22

**Comment by rdig** [comment]
> Also, I didn't understand what the bug was and how you fixed it. What am I missing?

Changed the index value.

See the original:
https://github.com/JoinColony/colonyDapp/blob/93c6c8413f7f850c1baae85aad4d7a9451d55456/src/modules/dashboard/sagas/actions.ts#L438-L477

If you notice, the annotation message tx hash the same `index` as create transaction, that was actually a bug the bug that we didn't catch initial _(because frankly, nobody pays attention to that number in the gas station :P)_

What I did was to change the last index, the one from the annotation tx, to `2`.

https://github.com/JoinColony/colonyDapp/blob/2c42748465198bcb88faa1f9cad7b350887edf6d/src/modules/dashboard/sagas/actions/mintTokens.ts#L58-L97

File: src/i18n/en.json [hunk 1]
```diff
@@ -123,7 +123,7 @@
     "transaction.ColonyClient.claimColonyFunds.title": "Claim Pending Transactions",
     "transaction.ColonyClient.claimColonyFunds.description": "Claim Pending Transactions",
     "transaction.ColonyClient.moveFundsBetweenPotsWithProofs.title": "Move Funds Between Pots",
```
[the remark this answers is not in the document]
Comment (rdig) [reply src/i18n/en.json:125]: Nope. Pots are pots they are not renamed.

_"Each domain has a pot where funds get deposited"_. So if we change the name of the domain to team, then _"Each team has a pot where funds get deposited"_.

But if we go by your logic, of renaming pots to teams as well, then we'll have _"Each team has a team where funds get deposited"_

[the remark this answers is not in the document]
Comment (chinins) [reply src/i18n/en.json:125]: makes sense!

**Comment by chinins** [comment]
Of course! 1,2,3 instead of 1,2,2. Thank you! I'm not very attentive today :)

**Review by alicjakujawa — CHANGES_REQUESTED** [review]
Hmm why this is still a draft? Im almost sure it was ready for CR ? 🤔 
But either way i was testing it and small bug start to showing:

Screenshot 2021-02-22 at 07 30 53
When you create new team, you won't see 
"Team name: [team name]" until you refresh page - it works fine earlier on

Apart of this small bug everything looks great, happy that we extract also sagas to separate files :)

**Comment by rdig** [comment]
> Hmm why this is still a draft? I'm almost sure it was ready for CR ? thinking

You are right. I forgot to switch it after I refactored the saga into their own files

> But either way I was testing it and a small bug start to show:

This is not something I introduced, this exists currently in production and can be replicated sporadically. It's due to IPFS failing to fetch the new name set for the domain when loading the list.

This will be fixed properly once we get to implementing the IPFS local cache DEV-171

**Review by alicjakujawa — APPROVED** [review]

**Review by ArmandoGraterol — APPROVED** [review]
It's lit :fire:

**Pull request merged**